src/SpectralDataDialog.cpp

```diff
@@ -0,0 +1,251 @@
+{
+   SetName();
+
+   mProject = parent;
+   mSelected = 0;
+   mAudioIOBusy = false;
+
+   //------------------------- Main section --------------------
+   // Construct the GUI.
+   ShuttleGui S(this, eIsCreating);
+   Populate(S);
+
+   wxTheApp->Bind(EVT_AUDIOIO_PLAYBACK,
+                  &SpectralDataDialog::OnAudioIO,
+                  this);
+
+   wxTheApp->Bind(EVT_AUDIOIO_CAPTURE,
+                  &SpectralDataDialog::OnAudioIO,
+                  this);
+
+   Clipboard::Get().Bind(
+         EVT_CLIPBOARD_CHANGE, &SpectralDataDialog::UpdateDisplay, this);
+   parent->Bind(EVT_UNDO_PUSHED, &SpectralDataDialog::UpdateDisplay, this);
+   parent->Bind(EVT_UNDO_MODIFIED, &SpectralDataDialog::UpdateDisplay, this);
+   parent->Bind(EVT_UNDO_OR_REDO, &SpectralDataDialog::UpdateDisplay, this);
+   parent->Bind(EVT_UNDO_RESET, &SpectralDataDialog::UpdateDisplay, this);
+   parent->Bind(EVT_UNDO_PURGE, &SpectralDataDialog::UpdateDisplay, this);
+}
+
+static const AudacityProject::AttachedWindows::RegisteredFactory key{
+      []( AudacityProject &project ){
+         return safenew SpectralDataDialog( &project );
+      }
+};
+
+void SpectralDataDialog::Populate(ShuttleGui & S)
+{
+   auto imageList = std::make_unique<wxImageList>(9, 16);
+   imageList->Add(wxIcon(empty9x16_xpm));
+   imageList->Add(wxIcon(arrow_xpm));
+
+   S.StartVerticalLay(true);
+   {
+      S.StartStatic(XO("Options"), 1);
+      {
+         S.Id(ID_CHECKBOX_OVERTONES)
+               .AddCheckBox(XXO("Enable overtones selection"), false),
+         S.Id(ID_CHECKBOX_SMART)
+               .AddCheckBox(XXO("Enable smart selection"), false);
+      }
+      S.EndStatic();
+
+      S.StartStatic(XO("Brush radius"), 1);
+      {
+         S.Id(ID_SLIDER_BRUSH_SIZE)
+               .Style(wxSL_HORIZONTAL)
+               .Name(XO("Custom brush size"))
+               .AddSlider( {}, 5, 10, 1);
+      }
+      S.EndStatic();
+   }
+   mApplyBtn = S.Id(ID_ON_APPLY)
+         .AddButton(XXO("Apply effect to selection."));
+   S.EndVerticalLay();
+   // ----------------------- End of main section --------------
+
+   Layout();
+   Fit();
+   SetMinSize(GetSize());
+}
+
+void SpectralDataDialog::OnAudioIO(wxCommandEvent& evt)
+{
+   evt.Skip();
+
+   if (evt.GetInt() != 0)
+      mAudioIOBusy = true;
+   else
+      mAudioIOBusy = false;
+}
+
+void SpectralDataDialog::UpdateDisplay(wxEvent& e)
+{
+   e.Skip();
+   if(IsShown())
+      DoUpdate();
+}
+
+bool SpectralDataDialog::Show( bool show )
+{
+   if ( show && !IsShown())
+      DoUpdate();
+   return wxDialogWrapper::Show( show );
+}
+
+void SpectralDataDialog::DoUpdate()
+{
+}
+
+void SpectralDataDialog::OnCloseWindow(wxCloseEvent & WXUNUSED(event))
+{
+   this->Show(false);
+}
+
+void SpectralDataDialog::OnShow(wxShowEvent & event)
+{
+   if (event.IsShown())
+   {
+//      mApplyBtn->SetFocus();
+   }
+}
+
+// PrefsListener implementation
+void SpectralDataDialog::UpdatePrefs()
+{
```

Comment:
Finally you may wonder if you really need this bit of copied event handling from History Window.

You do!

Because one thing that might change in Preferences is the language.  We have made sure that all non-modal dialogs re-create themselves so that user-visible strings are translated into the newly chosen language.  It is right to do here.  Of course you won't see translation if you try it now, becase your new intenationalized strings (those in the `XO` macro) are not yet entered in the catalogs that human translators have yet to fill in.